### PR TITLE
perf: avoid eagle draft seq_lens sync

### PR DESCRIPTION
**Pull request by peter941221**
Motivation
Issue #26171 reports that EAGLE-3 on Blackwell still has avoidable GPU->CPU synchronization in the draft preprocessing path.

One remaining sync point is in `EAGLEWorker._draft_preprocess_decode()`, where we compute `batch.seq_lens_sum` via `torch.sum(batch.seq_lens).item()`:

- `python/sglang/srt/speculative/eagle_worker.py:727`

That forces a scalar readback from the GPU tensor even though the decode batch already maintains a CPU mirror in `batch.seq_lens_cpu`.

The rest of the forward-batch setup already treats the CPU mirror as the canonical source for this sum:

- `python/sglang/srt/model_executor/forward_batch_info.py:493-495`

This is the same class of cleanup as #20266, but on the draft preprocess side instead of `eagle_info.verify()`.

Modifications
Replace the draft preprocess sum with `int(batch.seq_lens_cpu.sum())` so the code reuses the existing CPU-side sequence lengths instead of synchronizing the GPU tensor.

Preserved behavior
`ScheduleBatch` keeps `seq_lens` and `seq_lens_cpu` in lockstep during decode preparation and speculative batch filtering / merging, so this only changes where the scalar is read from, not the value.

Validation
- `python -m compileall python/sglang/srt/speculative/eagle_worker.py`
- diff inspection against `ForwardBatch.init_new()` to confirm the new source matches the existing lazy recomputation path in `python/sglang/srt/model_executor/forward_batch_info.py:493-495`

Checklist
- [ ] Format your code according to the contribution guide.
- [ ] Add/update tests if maintainers want a focused regression test for this path.
- [ ] Run broader speculative decoding performance validation on Blackwell hardware.

---
### CI States

Latest PR Test (Base): :x: [Run #26498214362](https://github.com/sgl-project/sglang/actions/runs/26498214362)
Latest PR Test (Extra): :x: [Run #26498214389](https://github.com/sgl-project/sglang/actions/runs/26498214389)

### CODE REVIEW

**Comment by peter941221** [comment]
Converted this PR to ready for review. CI is still blocked by the required run-ci label, and I do not have permission to add that label on this fork PR. Could a maintainer or another authorized contributor please tag run-ci?

**Comment by DLA-learn** [comment]
Thanks for working on this PR.

I opened #26171, and this change directly matches the GPU-CPU sync point I observed in the EAGLE-3 draft preprocessing path on Blackwell GPUs.

Replacing:

```python
torch.sum(batch.seq_lens).item()
```

with the CPU-side `seq_lens_cpu` avoids forcing a GPU-to-CPU synchronization inside `_draft_preprocess_decode()`. This is consistent with the analysis in #26171: the non-speculative baseline scaled well on Blackwell, but the EAGLE-3 speedup ratio was limited by overhead specific to the speculative path.

The benchmark numbers in #26171 were measured after removing this sync point and showed clear gains across gsm8k, math500, humaneval, and mtbench. From my side, this PR looks like the right fix for the issue.

I can help re-run the same Blackwell benchmark setup on this PR branch if needed. Could a maintainer please help trigger CI / add the required `run-ci` label?

**Comment by peter941221** [comment]
The current failures look like gate/trigger issues, not a concrete test failure from the patch. The red checks are on the gate/finalizer side while most underlying jobs are skipped. Could a maintainer add the required CI label or rerun the gated workflows?

**Comment by DLA-learn** [comment]
Thanks for checking this.

That matches my understanding as well. The current red checks look like gated CI trigger/finalizer issues rather than failures caused by the patch itself, since most underlying jobs were skipped.

@merrymercy @hnyls2002 Could you please help add the required `run-ci` label or rerun the gated workflows when you have a chance?

This PR directly addresses the GPU-CPU sync issue I reported in #26171, and I’m happy to help validate the Blackwell performance once CI is properly triggered.
